corehq/apps/data_pipeline_audit/dbacessors.py

```diff
@@ -130,44 +132,68 @@ def _get_sql_cases_by_doc_type(domain):
     return counter
 
 
-def get_sql_case_ids(domain, doc_type):
+def get_sql_case_ids(domain, doc_type, startdate, enddate):
     sql_ids = set()
     deleted = doc_type == 'CommCareCase-Deleted'
     for db_alias in get_sql_db_aliases_in_use():
         queryset = CommCareCaseSQL.objects.using(db_alias) \
-            .filter(domain=domain, deleted=deleted).values_list('case_id', flat=True)
-        sql_ids.update(list(queryset))
+            .filter(domain=domain, deleted=deleted)
+
+        if startdate:
+            queryset = queryset.filter(server_modified_on__gte=startdate)
+
+        if enddate:
+            queryset = queryset.filter(server_modified_on__lt=enddate)
+
+        sql_ids.update(list(queryset.values_list('case_id', flat=True)))
     return sql_ids
 
 
-def get_sql_form_ids(domain, doc_type):
+def get_sql_form_ids(domain, doc_type, startdate, enddate):
     sql_ids = set()
     state = doc_type_to_state[doc_type]
     for db_alias in get_sql_db_aliases_in_use():
         queryset = XFormInstanceSQL.objects.using(db_alias) \
-            .filter(domain=domain, state=state).values_list('form_id', flat=True)
-        sql_ids.update(list(queryset))
+            .filter(domain=domain, state=state)
+
+        if startdate:
+            queryset = queryset.filter(received_on__gte=startdate)
+
+        if enddate:
+            queryset = queryset.filter(received_on__lt=enddate)
+
+        sql_ids.update(list(queryset.values_list('form_id', flat=True)))
     return sql_ids
 
 
-def get_es_case_ids(domain, doc_type):
-    return _get_es_doc_ids(es.CaseES, domain, doc_type)
+def get_es_case_ids(domain, doc_type, startdate, enddate):
+    datefilter = None
+    if startdate or enddate:
+        datefilter = es.cases.modified_range(gte=startdate, lt=enddate)
```

Comment:
This should be server_modified_range to match sql

Reply:
Thanks: ec3c85d